src/core/built-in/rules/package-json-properties.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const path = require('path');
+
+const Rule = require('../../plugins/rule');
+const docs = require('../../../utils/docs');
+
+class PackageJsonProperties extends Rule {
+  constructor(context) {
+    super(context);
+
+    if (!this.context.options['required'] && !this.context.options['blacklist']) {
+      throw new Error(`The rule must have at least one option, "required" or "blacklist"`);
+    }
+
+    if (this.context.options['required'] && !Array.isArray(this.context.options['required'])) {
+      throw new Error(`Wrong "required" argument, an array is expected`);
+    }
+
+    if (this.context.options['blacklist'] && !Array.isArray(this.context.options['blacklist'])) {
+      throw new Error(`Wrong "blacklist" argument, an array is expected`);
+    }
+  }
+
+  run(sandbox) {
+    const packagejson = this._getPackageJsonPath();
+
+    const requiredProps = this.context.options['required'];
+    if (requiredProps) {
+      for (let index = 0; index < requiredProps.length; index++) {
+        const property = requiredProps[index];
+        if (packagejson[property] === undefined) {
+          sandbox.report({
+            message: `package.json is missing the required property "${property}"`
+          });
+          return;
```

Comment:
Here and below you are early returning, I think it may be more useful if we make a list of all the missing properties / black listed properties and send them all. That way the user can fix them all at once instead of fixing one, running the test again, fixing the next one, and so on

Reply:
Yeah, I thought about it, did it like this for performance, it's an early return

Reply:
I'm thinking it might be better to do something like this for a better user experience:
```
const missingProps = requiredProps.filter(prop => !packagejson.hasOwnProperty(prop));
if (missingProps.length > 0) {
   sandbox.report({
     message: `package.json is missing the required properties "${missingProps.join(',')}"`
   });
   return;
}
```
This way they see all the required properties at once instead of having to run it multiple times (in the case they are missing multiple required properties). This would apply to the other checks as well.